src/classes/UTIL_Describe.cls

```diff
@@ -370,4 +370,108 @@ public class UTIL_Describe {
         }
         return strFields.substring(0, strFields.length()-1);
     }
+
+    /*******************************************************************************************************
+    * @description Cache for the record type DeveloperName-to-ID map.
+    */
+    private static Map<String, Id> mapRecordTypes;
+    
+    /*******************************************************************************************************
+     * @description Returns a map of active, user-available RecordType IDs for a given SObjectType,
+     * keyed by each RecordType's unique, unchanging DeveloperName. 
+     * @param objectName Name of the object to retrieve record types for.
+     * @return Map<String, Id> Map of DeveloperName-to-ID of available record types for object specified. 
+     */
+    public static Map<String, Id> getRecTypesMapByDevName(String objectName) {	    
+	    return getRecTypesMap(objectName, true);
+	}
+	
+	/*******************************************************************************************************
+     * @description Returns a map of active, user-available RecordType IDs for a given SObjectType,
+     * keyed by each RecordType's Name. 
+     * @param objectName Name of the object to retrieve record types for.
+     * @return Map<String, Id> Map of Name-to-ID of available record types for object specified. 
+     */
+    @AuraEnabled
+    public static Map<String, Id> getRecTypesMapByName(String objectName) {
+        return getRecTypesMap(objectName, false);
+    }
+    
+    /*******************************************************************************************************
+     * @description Returns a map of active, user-available RecordType IDs for a given SObjectType, keyed
+     * by either the RecordType's Name or the Developer Name, depending on the value of the second param. 
+     * @param objectName Name of the object to retrieve record types for.
+     * @param devName True if we want the key of the returned map to be the record type's developer name.
+     * False if we want it to be the friendly name.
+     * @return Map<String, Id> Map of Name-to-ID or DeveloperName-to-ID (depending on the second paramater), 
+     * of available record types for object specified. 
+     */
+    private static Map<String, Id> getRecTypesMap(String objectName, Boolean devName) {      
+        if(mapRecordTypes == null) {
```

Comment:
Not sure I follow. Are you thinking of record types that are added or removed while the transaction that called this method is still running? It seems like a rare scenario to me. I don't think record types are added or removed very often. And right now the only ones that call these methods are the install script and the settings pages...